static/app/utils/replays/replayReader.tsx

```diff
@@ -48,31 +49,48 @@ export default class ReplayReader {
    * The original `this._event.startTimestamp` and `this._event.endTimestamp`
    * are the same. It's because the root replay event is re-purposing the
    * `transaction` type, but it is not a real span occuring over time.
-   * So we need to figure out the real end time, in milliseconds.
+   * So we need to figure out the real start and end timestamps based on when
+   * first and last bits of data were collected. In milliseconds.
    */
-  private _getEndTimestampMS() {
+  private _getTimestampsMS = memoize(() => {
     const crumbs = this.getRawCrumbs();
     const spans = this.getRawSpans();
 
-    const lastRRweb = last(this._rrwebEvents);
-    const lastBreadcrumb = last(crumbs);
-    const lastSpan = last(spans);
-
-    return Math.max(
-      lastRRweb?.timestamp || 0,
-      +new Date(lastBreadcrumb?.timestamp || 0),
-      (lastSpan?.timestamp || 0) * 1000
+    // Sort our data so we know we're getting the first or last record from the array.
+    const sortedRRwebEvents = Array.from(this._rrwebEvents).sort(
+      (a, b) => a.timestamp - b.timestamp
     );
-  }
+    const sortedCrumbs = Array.from(crumbs).sort(
+      (a, b) => +new Date(a?.timestamp || 0) - +new Date(b?.timestamp || 0)
+    );
+    const startSortedSpans = Array.from(spans).sort(
+      (a, b) => a.start_timestamp - b.start_timestamp
+    );
```

Comment:
Should this be moved to the `getRaw*` methods? I would expect this data to always be sorted.

Reply:
Spans are the special case, I'm sorting here both by the start and also end timestamps. But yeah i guess sorting by start is a sane default to guarentee